File: sorting/quickSelect.js
const select = (array, left, right, k) => {
  if (left == right)
    // If the array contains only one element,
    return array[left]; // return that element
  const pivotIndex = partition(array, left, right);
  // The pivot is in its final sorted position
  if (k == pivotIndex) return array[k];
  else if (k < pivotIndex) return select(array, left, pivotIndex - 1, k);
  else return select(array, pivotIndex + 1, right, k);
};
const partition = (array, left, right) => {
  const pivotIndex = Math.floor((left + right) / 2);
  const pivot = array[pivotIndex];
  swap(array, pivotIndex, right);
  let storeIndex = left;
  for (let i = left; i <= right; i++) {
    if (array[i] < pivot) {
      swap(array, i, storeIndex);
      storeIndex++;
    }
  }
  swap(array, right, storeIndex);
  return storeIndex;
};

const swap = (array, i, j) => {
  const temp = array[i];
  array[i] = array[j];
  array[j] = temp;
};

const findKthLargest = (array, k) => {
  k = array.length - k;
  return select(array, 0, array.length - 1, k);
};

File: sorting/quickSelect.py
from typing import List

def select(nums: List[int], left: int, right: int, k: int) -> int:
  pivotIndex = partition(nums, left, right)
  # The pivot is in its final sorted position
  if k == pivotIndex: return nums[k]
  elif k < pivotIndex: return select(nums, left, pivotIndex - 1, k)
  else: return select(nums, pivotIndex + 1, right, k)

def partition(nums: List[int], left: int, right: int) -> int:
  mid = (left + right) // 2
  pivot = nums[mid]
  nums[mid], nums[right] = nums[right], nums[mid]
  for i in range(left, right + 1):
    if nums[i] < pivot:
      nums[i], nums[left] = nums[left], nums[i] 
      left += 1
  nums[right], nums[left] = nums[left], nums[right]
  return left

def findKthLargest(nums: List[int], k: int) -> int:
  n = len(nums)
  k = n - k
  return select(nums, 0, n - 1, k)